src/sql/services/notebook/notebookRegistry.ts

```diff
@@ -54,42 +55,28 @@ let notebookContrib: IJSONSchema = {
 };
 
 export interface INotebookProviderRegistry {
+	readonly providers: NotebookProviderDescription[];
+	readonly onNewProvider: Event<{ id: string, properties: NotebookProviderDescription }>;
+
 	registerNotebookProvider(provider: NotebookProviderDescription): void;
-	getSupportedFileExtensions(): string[];
-	getProviderForFileType(fileType: string): string;
 }
 
 class NotebookProviderRegistry implements INotebookProviderRegistry {
 	private providerIdToProviders = new Map<string, NotebookProviderDescription>();
-	private fileToProviders = new Map<string, NotebookProviderDescription>();
+	private _onNewProvider = new Emitter<{ id: string, properties: NotebookProviderDescription }>();
+	public readonly onNewProvider: Event<{ id: string, properties: NotebookProviderDescription }> = this._onNewProvider.event;
 
 	registerNotebookProvider(provider: NotebookProviderDescription): void {
 		// Note: this method intentionally overrides default provider for a file type.
 		// This means that any built-in provider will be overridden by registered extensions
 		this.providerIdToProviders.set(provider.provider, provider);
-		if (provider.fileExtensions) {
-			if (Array.isArray<string>(provider.fileExtensions)) {
-				for (let fileType of provider.fileExtensions) {
-					this.addFileProvider(fileType, provider);
-				}
-			} else {
-				this.addFileProvider(provider.fileExtensions, provider);
-			}
-		}
-	}
-
-	private addFileProvider(fileType: string, provider: NotebookProviderDescription) {
-		this.fileToProviders.set(fileType.toUpperCase(), provider);
-	}
-
-	getSupportedFileExtensions(): string[] {
-		return Array.from(this.fileToProviders.keys());
+		this._onNewProvider.fire( { id: provider.provider, properties: provider });
```

Comment:
This is super confusing. We don't need to fix this now, but crap... Just looking at this, I have no idea what the difference is between 'provider.provider' and 'provider'.

Reply:
Yeah... agreed. Will see if I can fix